api/src/main/java/org/apache/iceberg/encryption/EncryptionManager.java

```diff
@@ -47,11 +49,26 @@ public interface EncryptionManager extends Serializable {
    * By default this calls the single-file decryption method for each element in the iterator.
    * Implementations can override this for a variety of optimizations. For example, an
    * implementation can perform lookahead on the input iterator and fetch encryption keys in batch.
+   *
+   * @deprecated Use {@link #decrypt(Iterator)}.
    */
+  @Deprecated
```

Comment:
This is a public API. We shouldn't make changes to a public API just because it is being called incorrectly in some places.